Basic/ClientDriven/Assets/Scripts/NetworkObjectSpawner.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using Unity.Netcode;
+using UnityEngine;
+using UnityEngine.Networking;
+using Random = System.Random;
+
+namespace Unity.Multiplayer.Samples.ClientDriven
+{
+    /// <summary>
+    /// Spawn a NetworkObject at this transform's position when NetworkManer's server is started.
+    /// </summary>
+    /// <remarks>
+    /// A NetworkManager is expected to be part of the scene that this NetworkObject is a part of.
+    /// </remarks>
+    internal class NetworkObjectSpawner : MonoBehaviour
+    {
+        [SerializeField]
+        private NetworkObject prefabReference;
+        [SerializeField]
+        private NetworkManager m_NetworkManager;
+
+        void Start()
+        {
+            Debug.Assert(m_NetworkManager != null, "The NetworkManager needs to be referenced!");
+            if (m_NetworkManager == null)
+            {
+                return;
+            }
+
+            m_NetworkManager.OnServerStarted += OnServerStartedIngredientSpawn;
+        }
+
+        void OnDestroy()
+        {
+            m_NetworkManager.OnServerStarted -= OnServerStartedIngredientSpawn;
+        }
+
+        void OnServerStartedIngredientSpawn()
+        {
+            Random randomGenerator = new Random();
```

Comment:
This and line 42 you can convert to var, since the types are explicit.

Reply:
It was a var before but I converted it to a Random as it was a suggestion from Paolo, would there be an advantage through turning it back to a var? Otherwise I would keep it as it is

Reply:
Per the code standard doc:
`'var' required on any 'new' where the type we want is the same as what is being constructed`

It's necessary for line 40, and I would argue _recommended_, though not necessary, for line 42.